mmpose/datasets/pipelines/hand_transform.py

```diff
@@ -0,0 +1,198 @@
+import numpy as np
+
+from mmpose.core.post_processing import fliplr_joints
+from mmpose.datasets.pipelines import Compose
+from mmpose.datasets.registry import PIPELINES
+
+
+@PIPELINES.register_module()
+class HandRandomFlip:
+    """Data augmentation with random image flip.
+
+    Required keys: 'img', 'joints_3d', 'joints_3d_visible', 'center',
+    'hand_type', 'rel_root_depth' and 'ann_info'.
+    Modifies key: 'img', 'joints_3d', 'joints_3d_visible', 'center',
+    'hand_type', 'rel_root_depth'.
+
+    Args:
+        flip_prob (float): Probability of flip.
+    """
+
+    def __init__(self, flip_prob=0.5):
+        self.flip_prob = flip_prob
+
+    def __call__(self, results):
+        """Perform data augmentation with random image flip."""
+        img = results['img']
+        joints_3d = results['joints_3d']
+        joints_3d_visible = results['joints_3d_visible']
+        center = results['center']
+        hand_type = results['hand_type']
+        rel_root_depth = results['rel_root_depth']
+
+        if np.random.rand() <= self.flip_prob:
+            img = img[:, ::-1, :]
+
+            joints_3d, joints_3d_visible = fliplr_joints(
+                joints_3d, joints_3d_visible, img.shape[1],
+                results['ann_info']['flip_pairs'])
+            center[0] = img.shape[1] - center[0] - 1
+            hand_type[0], hand_type[1] = hand_type[1].copy(
+            ), hand_type[0].copy()
+            rel_root_depth = -rel_root_depth
+
+        results['img'] = img
+        results['joints_3d'] = joints_3d
+        results['joints_3d_visible'] = joints_3d_visible
+        results['center'] = center
+        results['hand_type'] = hand_type
+        results['rel_root_depth'] = rel_root_depth
+
+        return results
+
+
+@PIPELINES.register_module()
+class HandGetRandomTranslation:
```

Comment:
Also applicable to Body right?

Reply:
Yes. I have moved it to topdown_transform